modules/spring-data-ext/src/main/java/org/apache/ignite/springdata/repository/support/IgniteRepositoryFactoryBean.java

```diff
@@ -81,12 +84,30 @@ protected IgniteRepositoryFactoryBean(Class<? extends T> repositoryInterface) {
                     return new IgniteRepositoryFactory(path);
                 }
                 catch (BeansException ex3) {
-                    throw new IgniteException("Failed to initialize Ignite repository factory. Ignite instance or" +
-                        " IgniteConfiguration or a path to Ignite's spring XML configuration must be defined in the" +
-                        " application configuration");
+                    throw new IgniteException("Failed to initialize Ignite repository factory. One of the following" +
+                        " beans must be defined in application configuration: \"igniteInstance\", \"igniteCfg\"," +
+                        " \"igniteSpringCfgPath\".");
                 }
             }
         }
     }
+
+    /**
+     * Creates instance of {@link IgniteRepositoryFactory} via reflection.
+     *
+     * @param args Arguments to be used when creating the {@link IgniteRepositoryFactory} instance.
+     * @return {@link IgniteRepositoryFactory} instance.
+     */
+    private IgniteRepositoryFactory createRepositoryFactory(Object... args) {
+        Constructor<?> ctor = ReflectionUtils.findConstructor(IgniteRepositoryFactory.class, args);
```

Comment:
Let's use if/else if instead of reflection. 
Or better inline these if's to `createRepositoryFactory()` method (like in other modules).

Reply:
Done.